product/interfaces/graphql/product/dto/variantsToVariationSelectionsMapper.go

```diff
@@ -134,53 +167,48 @@ func (m *variantsToVariationSelectionsMapper) hasVariantsWithMatchingAttributes(
 
 func (m *variantsToVariationSelectionsMapper) buildVariationSelectionOptions(attributeGroup *attributeGroup) []VariationSelectionOption {
 	var options []VariationSelectionOption
-	attributeLabelsInOrder, sortKeyExists := m.variationAttributesSorting[attributeGroup.Code]
-
-	if sortKeyExists {
-		for _, attributeLabel := range attributeLabelsInOrder {
-			attribute := attributeGroup.getAttributeByLabel(attributeLabel)
-
-			if attribute != nil {
-				var state VariationSelectionOptionState
-				var marketPlaceCode string
-
-				if m.hasActiveVariant() {
-					mergedAttributes := m.getActiveAttributesWithOverwrite(*attribute)
-					exactMatchingVariant := m.findMatchingVariant(mergedAttributes)
-
-					if exactMatchingVariant != nil {
-						if exactMatchingVariant.MarketPlaceCode == m.activeVariant.MarketPlaceCode {
-							state = VariationSelectionOptionStateActive
-							marketPlaceCode = exactMatchingVariant.MarketPlaceCode
-						} else {
-							state = VariationSelectionOptionStateMatch
-							marketPlaceCode = exactMatchingVariant.MarketPlaceCode
-						}
-					} else {
-						fallbackVariant := m.findMatchingVariant([]domain.Attribute{*attribute})
-
-						if fallbackVariant != nil {
-							state = VariationSelectionOptionStateNoMatch
-							marketPlaceCode = fallbackVariant.MarketPlaceCode
-						}
-					}
+
+	attributeGroup.eachAttributeInOrder(func(attribute *domain.Attribute) {
+		var state VariationSelectionOptionState
+		var matchingVariant domain.Variant
+
+		if m.hasActiveVariant() {
+			mergedAttributes := m.getActiveAttributesWithOverwrite(*attribute)
+			exactMatchingVariant := m.findMatchingVariant(mergedAttributes)
+
+			if exactMatchingVariant != nil {
```

Comment:
Is all the if/else branching necessary? looks like a lot of cognitive load to me. is it possible to decrease that using early returns/always doing the else part?

Reply:
Yeah I know, i´ts the most ugly method in this whole class because the logic behind it is inherently complex. I could at least try to split it up in multiple methods to make it a bit more readable

Reply:
I got rid of some ifs by separating the logic into smaller methods. I think now it´s better to grasp what is getting on.

Reply:
Jeah way better! Added a small improvement, will merge when pipeline is green. 👍